PowerEditor/src/TinyXml/tinyxml.cpp

```diff
@@ -551,20 +551,37 @@ void TiXmlElement::SetAttribute( const TCHAR * name, const TCHAR * _value )
 	}
 }
 
-void TiXmlElement::Print( FILE* cfile, int depth ) const
+const char one_ws[] = " ";
+const char four_ws[] = "    ";
+const char eol[] = "\r\n";
+
+const generic_string tagOpenSymb = TEXT("<");
+const generic_string tagOpenSlash = TEXT("</");
+const generic_string tagCloseSymb = TEXT(">");
+
+const char tagCloseChar[] = ">";
+const char tagCloseSymbFinal[] = " />";
+
+void TiXmlElement::Print( CFile& cfile, int depth ) const
 {
 	int i;
 	for ( i=0; i<depth; i++ )
 	{
-		generic_fprintf( cfile, TEXT("    ") );
+		cfile.Write(four_ws, 4);
+		//generic_fprintf( cfile, TEXT("    ") );
 	}
 
-	generic_fprintf( cfile, TEXT("<%ls"), value.c_str() );
+	generic_string tagOpenValue = tagOpenSymb + value;
+	std::string tagOpenValueA = wstring2string(tagOpenValue, CP_UTF8);
+	cfile.Write(tagOpenValueA.c_str(), static_cast<unsigned long>(tagOpenValueA.length()));
```

Comment:
I'm not sure you have to convert the string to ASCII.
Since, without the change `fwprintf` was used.

Reply:
https://stackoverflow.com/questions/28618715/c-writefile-unicode-characters